tip.lic

```diff
@@ -0,0 +1,99 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#tip
+=end
+
+# Background: I've always been annoyed by the need to specify tips in coppers.
+# DR's required syntax leaves you counting zeroes if you want to tip someone.
+# So a tip of 10 platinum kronars requires you to tip 100000 kronars. How many
+# zeroes is that? I lost count already. This simple script allows you to tip 
+# someone, in a similar syntax, but use the denomination to do so (e.g. "platinum").
+# <;tip gildaren 10 plat kron> now works. If you don't specify a currency,
+# this script also chooses the currency of the greatest coin you have on hand.
+
+custom_require.call %w[common common-money]
+
+class Tip
+
+  # include DRC for the message function
+  # include DRCM for use of converting amounts to copper
+  
+  def initialize
+    arg_definitions = [
+      [
+        { name: 'recipient', regex: /^[A-z\.\s\-']+$/i, variable: true, description: 'The recipent of your tip. Abbreviations ok.' },
+        { name: 'amount', regex: /^\d+$/i, variable: true, description: 'Numeric amount to tip, no commas (e.g. the number "100" in 100 kronars)' },
+        { name: 'denomination', regex: /(\bp(l|la|lat|lati|latin|latinu|latinum)?\b)|(\bg(o|ol|old)?\b)|(\bs(i|il|ilv|ilve|ilver)?\b)|(\bb(r|ro|ron|ronz|ronze)?\b)|(\bc(o|op|opp|oppe|opper)?\b)/i, variable: true, description: 'The denomination. Abbrevations ok. (e.g. plat, gold, silv)' },
+        { name: 'currency', regex: /(\bk(r|ro|ron|rona|ronar|ronars)?\b)|(l(i|ir|iru|irum|irums)?\b)|(\bd(o|ok|oko|okor|okora|okoras)?\b)/i, optional: true, variable: true, description: 'Optional currency. Abbreviations ok: kron, lirums, dokor' }         
+      ]
+    ]
+    args = parse_args(arg_definitions)  
+    gratuity(args.recipient, args.amount, args.denomination, args.currency)
+
+  end
+
+  def gratuity(recipient, amount, denomination, currency)
+    
+    total = convert_total(amount, denomination)
+    currency = handle_currency(total, currency)
+    
+    fput("tip #{recipient} #{total} #{currency}")
```

Comment:
Can this be a bput?

Something like this https://github.com/rpherbig/dr-scripts/blob/acc89c9a453fcb5afe06e4fa36e31e8b7b8bd4dd/bankbot.lic#L108

Reply:
It could be a bput, yes. And it originally was. Then I started handling the failure cases and realized that for many of them, the game already handles it with appropriate output, like if it can't find a recipient, or you don't have enough money to tip the desired amount.

But, I may have missed the other advantages of bput -- like that it'll wait for roundtime and recover from some other failure types and retry.

Reply:
In general, bput is designed to be a replacement for fput

Reply:
Gotcha. Will update.

Reply:
Now deployed as a bput.